core/domain/question_services.py

```diff
@@ -201,3 +203,39 @@ def update_question(committer_id, question_id, change_list, commit_message):
     """
     question = apply_change_list(question_id, change_list)
     _save_question(committer_id, question, change_list, commit_message)
+
+
+def get_questions_batch(
+        collection_id, skill_ids, user_id, question_play_counts, batch_size):
+    """Fetches a batch of questions.
+
+    Args:
+        collection_id: str. Id of the collection.
+        skill_ids: list(str). A list of skill ids.
```

Comment:
FYI, I don't understand the significance of this, which is making it a bit hard for me to understand how this function works.

Reply:
Even I find it weird in the doc, the function takes a list of skills and compares them with user skills list. Finally, The common skills of skills list and user skills are used for getting questions, which are associated to skills in common skills.
Wouldn't it be easier to give the user a bunch of question based on collection id?

Reply:
Oh, I think I know what's going on. We don't want to give the learner questions that they don't have the knowledge to attempt. So that's why it's got to be a subset of acquired skills.

Also, the user may only want to practice certain skills, or we might want to get a set of questions relevant to skills that they're weak in. This choice of skills to practice would be represented by the passed-in set of skill ids.

Reply:
Understood. Makes sense.